lib/facter/self_service.rb

```diff
@@ -87,4 +87,21 @@
     postgresversion = PuppetSelfService.pe_postgres_service_name
     { S0011: PuppetSelfService.service_running_enabled(postgresversion.to_s) }
   end
+
+  chunk(:S0012) do
+    next unless PuppetSelfService.replica? || PuppetSelfService.postgres? || PuppetSelfService.primary? || PuppetSelfService.compiler? || PuppetSelfService.legacy_compiler?
+    # Did Puppet Produce a report in the last run interval
+    summary_path = Puppet.settings['lastrunfile']
+    lastrunfile = YAML.load_file(summary_path)
```

Comment:
This code should probably handle a failure where the last run file doesn't exist. We can check that the file exists before loading it: `File.exist?(summary_path)` or a `begin/rescue` block that catches an `Errno::ENOENT`.